joblib/numpy_pickle.py

```diff
@@ -1,224 +1,231 @@
+        for fortran order.
+    dtype: numpy.ndarray dtype
+        Determine the data type of the wrapped array.
+    allow_mmap: bool
+        Determine if memory mapping is allowed on the wrapped array.
+        Default: False.
     """
-    magic = file_handle.read(len(_ZFILE_PREFIX))
-    # Pickling needs file-handles at the beginning of the file
-    file_handle.seek(0)
-    return magic
 
+    def __init__(self, subclass, shape, order, dtype, allow_mmap=False):
+        """Constructor. Store the useful information for later."""
+        self.subclass = subclass
+        self.shape = shape
+        self.order = order
+        self.dtype = dtype
+        self.allow_mmap = allow_mmap
 
-def read_zfile(file_handle):
-    """Read the z-file and return the content as a string
+    def write_array(self, array, pickler):
+        """Write array bytes to pickler file handle.
 
-    Z-files are raw data compressed with zlib used internally by joblib
-    for persistence. Backward compatibility is not guaranteed. Do not
-    use for external purposes.
-    """
-    file_handle.seek(0)
-    assert _read_magic(file_handle) == _ZFILE_PREFIX, \
-        "File does not have the right magic"
-    header_length = len(_ZFILE_PREFIX) + _MAX_LEN
-    length = file_handle.read(header_length)
-    length = length[len(_ZFILE_PREFIX):]
-    length = int(length, 16)
-
-    # With python2 and joblib version <= 0.8.4 compressed pickle header is one
-    # character wider so we need to ignore an additional space if present.
-    # Note: the first byte of the zlib data is guaranteed not to be a
-    # space according to
-    # https://tools.ietf.org/html/rfc6713#section-2.1
-    next_byte = file_handle.read(1)
-    if next_byte != b' ':
-        # The zlib compressed data has started and we need to go back
-        # one byte
-        file_handle.seek(header_length)
-
-    # We use the known length of the data to tell Zlib the size of the
-    # buffer to allocate.
-    data = zlib.decompress(file_handle.read(), 15, length)
-    assert len(data) == length, (
-        "Incorrect data length while decompressing %s."
-        "The file could be corrupted." % file_handle)
-    return data
-
-
-def write_zfile(file_handle, data, compress=1):
-    """Write the data in the given file as a Z-file.
-
-    Z-files are raw data compressed with zlib used internally by joblib
-    for persistence. Backward compatibility is not guarantied. Do not
-    use for external purposes.
-    """
-    file_handle.write(_ZFILE_PREFIX)
-    length = hex_str(len(data))
-    # Store the length of the data
-    file_handle.write(asbytes(length.ljust(_MAX_LEN)))
-    file_handle.write(zlib.compress(asbytes(data), compress))
+        This function is an adaptation of the numpy write_array function
+        available in version 0.10 in numpy/lib/format.py.
+        We added some code to support versions of numpy prior to 1.9.
```

Comment:
Is that still true? It doesn't seem to be the case looking at numpy.lib.format.write_array but I may be missing something.